restore-sql-server-transaction-logs/Function/main.py

```diff
@@ -0,0 +1,274 @@
+import functions_framework
+import google.auth
+import google.auth.transport.requests
+import googleapiclient.discovery
+import logging
+import os
+from time import sleep
+from google.cloud import storage
+from google.cloud import error_reporting
+
+
+def get_environment_variables() -> [bool, str, str, str, int, int, int]:
+    
+    err_msg=""
+
+    if not os.environ.get('USE_FIXED_FILE_NAME_FORMAT'):
+        err_msg = "Environment variable USE_FIXED_FILE_NAME_FORMAT is None or empty."
+        return False, err_msg, "", "", 0, 0, 0
+
+    if os.environ.get('USE_FIXED_FILE_NAME_FORMAT').lower() not in ["true", "false"]:
+        err_msg = "Environment variable USE_FIXED_FILE_NAME_FORMAT must be either True or False."
+        return False, err_msg, "", "", 0, 0, 0
+        
+
+    use_fixed_file_name_format = os.environ.get('USE_FIXED_FILE_NAME_FORMAT')
+    processed_bucket_name = "" if os.environ.get('MAX_REQUEST_ATTEMPTS') is None else os.environ.get('PROCESSED_BUCKET_NAME')
+    max_request_attempts = 5 if os.environ.get('MAX_REQUEST_ATTEMPTS') is None else int(os.environ.get('MAX_REQUEST_ATTEMPTS'))
+    max_request_fetch_time_seconds = 30 if os.environ.get('MAX_REQUEST_FETCH_TIME_SECONDS') is None else int(os.environ.get('MAX_REQUEST_FETCH_TIME_SECONDS'))
+    max_operation_fetch_time_seconds = 30 if os.environ.get('MAX_OPERATION_FETCH_TIME_SECONDS') is None else int(os.environ.get('MAX_OPERATION_FETCH_TIME_SECONDS'))
+
+    return True, err_msg, processed_bucket_name, use_fixed_file_name_format, max_request_attempts, max_request_fetch_time_seconds, max_operation_fetch_time_seconds
+
+
+def extract_info_from_file_name(object_name: str) -> [bool, str, str, str, str, str]:
+
+    err_message=""
+    file_name=(object_name.split("/")[-1])
+    name_groups = file_name.split("_")
+
+    if len(name_groups)<2:
+        err_message = "The file name does not respect the imposed format <cloudsql_instance_name>_<database_name>*.*"
+        return False, err_message,  "", "", "", ""

```

Comment:
This is a somewhat convoluted way of handling error conditions. You should throw an appropriate exception instead of returning a complex object that you then need to handle.

Reply:
LGTM